sdk/storage/storage-queue/review/storage-queue.api.ts

```diff
@@ -0,0 +1,496 @@
+  // (undocumented)
+  readonly messageId: string;
+  // (undocumented)
+  readonly queueName: string;
+  update(popReceipt: string, message: string, visibilityTimeout?: number, options?: MessageIdUpdateOptions): Promise<Models.MessageIdUpdateResponse>;
+}
+
+// @public
+interface MessageIdDeleteOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface MessageIdUpdateOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface MessagesClearOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+class MessagesClient extends StorageClient {
+  constructor(connectionString: string, queueName: string, options?: NewPipelineOptions);
+  clear(options?: MessagesClearOptions): Promise<Models.MessagesClearResponse>;
+  dequeue(options?: MessagesDequeueOptions): Promise<MessagesDequeueResponse>;
+  enqueue(messageText: string, options?: MessagesEnqueueOptions): Promise<MessagesEnqueueResponse>;
+  getMessageIdClient(messageId: string): MessageIdClient;
+  peek(options?: MessagesPeekOptions): Promise<MessagesPeekResponse>;
+  // (undocumented)
+  readonly queueName: string;
+}
+
+// @public
+interface MessagesDequeueOptions extends Models.MessagesDequeueOptionalParams, CommonOptions {
+}
+
+// @public
+interface MessagesEnqueueOptions extends Models.MessagesEnqueueOptionalParams, CommonOptions {
+}
+
+// @public
+interface MessagesPeekOptions extends Models.MessagesPeekOptionalParams, CommonOptions {
+}
+
+// @public (undocumented)
+
+// @public
+export function newPipeline(credential: SharedKeyCredential | AnonymousCredential | TokenCredential, pipelineOptions?: NewPipelineOptions): Pipeline;
+
+// @public
+interface NewPipelineOptions {
+  httpClient?: IHttpClient;
+  keepAliveOptions?: KeepAliveOptions;
+  logger?: IHttpPipelineLogger;
+  // (undocumented)
+  proxy?: ProxySettings | string;
+  retryOptions?: RetryOptions;
+  telemetry?: TelemetryOptions;
+}
+
+// @public
+class Pipeline {
+  constructor(factories: RequestPolicyFactory[], options?: PipelineOptions);
+  readonly factories: RequestPolicyFactory[];
+  readonly options: PipelineOptions;
+  toServiceClientOptions(): ServiceClientOptions;
+}
+
+// @public
+interface PipelineOptions {
+  HTTPClient?: IHttpClient;
+  logger?: IHttpPipelineLogger;
+}
+
+// @public
+class QueueClient extends StorageClient {
+  constructor(connectionString: string, queueName: string, options?: NewPipelineOptions);
+  create(options?: QueueCreateOptions): Promise<Models.QueueCreateResponse>;
+  delete(options?: QueueDeleteOptions): Promise<Models.QueueDeleteResponse>;
+  getAccessPolicy(options?: QueueGetAccessPolicyOptions): Promise<QueueGetAccessPolicyResponse>;
+  getMessagesClient(): MessagesClient;
+  getProperties(options?: QueueGetPropertiesOptions): Promise<Models.QueueGetPropertiesResponse>;
+  // (undocumented)
+  readonly queueName: string;
```

Comment:
>queueName [](start = 11, length = 9)

Be careful for the implementation of `queueName` for emulator style urls like `[http://127.0.0.1:10000/devstoreaccount1/queuename`](http://127.0.0.1:10000/devstoreaccount1/queuename`)

Reply:
We have a bug to track this.

Reply:
#5446